### PR TITLE
specify executor

### PR DESCRIPTION
**Pull request by teaey**
Some times, we need events to be executed by specified executor.

### CODE REVIEW

File: guava/src/com/google/common/eventbus/SubscriberRegistry.java [hunk 1]
```diff
@@ -174,6 +199,20 @@ void unregister(Object listener) {
     return methodsInListener;
   }
 
+  /**
+   * Returns all subscribers for the given listener grouped by the type of event they subscribe to.
+   */
+  private Multimap<Class<?>, Subscriber> findAllSubscribers(Object listener, Executor executor) {
```
Comment (liach) [inline guava/src/com/google/common/eventbus/SubscriberRegistry.java]: This can be just a `Map` since you do not need `Multimap` functionalities.

**Comment by teaey** [comment]
@cpovirk @kluever @lowasser 
Can anybody accept this PR?

File: guava/src/com/google/common/eventbus/SubscriberRegistry.java (hunk 1 above)
Comment (teaey) [reply guava/src/com/google/common/eventbus/SubscriberRegistry.java]: Keep the same semantics as the original method, eg:
```java
private Multimap<Class<?>, Subscriber> findAllSubscribers(Object listener)
```

Comment (teaey) [reply guava/src/com/google/common/eventbus/SubscriberRegistry.java]: @liach It's okay?

Comment (liach) [reply guava/src/com/google/common/eventbus/SubscriberRegistry.java]: Yeah, you can ignore my old comment.

**Comment by teaey** [comment]
@liach 👍  Can you accept my PR now？

**Comment by jbduncan** [comment]
@teaey You'll have to wait for someone from the Guava team to have a look at this PR first, as neither @liach nor myself are on the Guava team nor employees of Google ourselves.

Did you read the [API](https://github.com/google/guava/blob/master/CONTRIBUTING.md#api-changes) section of the contribution guidelines? :)

**Comment by jbduncan** [comment]
@teaey That's correct.

It also means that Google employees have very high standards for the APIs they accept into Guava and follow a rigorous internal review process, so this PR may be rejected initially and they'll ask you to rewrite the code, or they may decide that the API doesn't really add anything and outright reject it.

But if either of those things do happen to you, try not to be discouraged, for it would actually be a very good thing as it would mean that Guava's API remains relatively easy to use and of high quality. :)

**Comment by liach** [comment]
google doesn't really like additions. In comparison, bug fixes get accepted more easily.

**Comment by nglorioso** [comment]
Realistically, we're unlikely to add functionality to EventBus at this point.

**Pull request closed without merging**